doc/developer/design/20231110_aws_connections.md

```diff
@@ -107,16 +112,54 @@ When creating the connection, users should specify `WITH (VALIDATE = false)` in
 This is because we won't be able to validate the connection unless the trust policy has
 been added on the user's end.
 
+#### `mz_aws_connections` table
+
+We'll add a new `mz_aws_connections` table with the following definition:
+
+```sql
+CREATE TABLE mz_internal.mz_aws_connections (
+    -- The ID of the connection in `mz_connections`.
+    id text NOT NULL,
+    -- The value of the `ENDPOINT` option, if specified. `NULL` otherwise.
+    endpoint text,
+    -- The value of the `REGION` option, if specified. `NULL` otherwise.
+    region text,
+    -- The value of the `AWS ACCESS KEY ID` option, if specified. `NULL` otherwise.
+    access_key_id text,
+    -- The value of the `ASSUME ROLE ARN` option, if specified. `NULL` otherwise.
+    assume_role_arn text,
+    -- The value of the `ASSUME ROLE SESSION ARN` option, if specified. `NULL` otherwise.
+    assume_role_session_name text,
+    -- The AWS IAM principal Materialize will use to assume the role, if
+    -- the connection configured `ASSUME ROLE ARN`. `NULL` otherwise.
+    principal text,
+    -- The external ID Materialize will use to assume the role, if
+    -- the connection configured `ASSUME ROLE ARN`. `NULL` otherwise.
+    external_id text,
+    -- A fully rendered AWS IAM trust policy that should be installed on the
+    -- role specified by `ASSUME ROLE ARN` to grant Materialize permission to
+    -- assume the role. `NULL` if `ASSUME ROLE ARN` is not specified.
+    trust_policy jsonb
```

Comment:
See prior discussion [here](https://github.com/MaterializeInc/materialize/pull/23173/files#r1403734584) on whether we should include this field or not.

Reply:
OOC is it possible for a customer to have multiple statements in their trust policy, of which one of them is allowing AssumeRole to the corresponding External ID?

Reply:
Yes, possible! Should we call this `sample_trust_policy` or `example_trust_policy`, to make it clear that it's just a starting point?

Reply:
Changed the name to `example_trust_policy`.

Reply:
+1 to `example_trust_policy`, the name clarifies it might work out of the box for the straightforward cases or it might require some tinkering.